judge/views/problem.py

```diff
@@ -213,8 +226,117 @@ def get_context_data(self, **kwargs):
                                           context['description'], 'problem')
         context['meta_description'] = self.object.summary or metadata[0]
         context['og_image'] = self.object.og_image or metadata[1]
+
+        context['can_vote'] = self.can_vote(user, self.object)  # if this problem is votable by this user
+        # the vote this user has already cast on this problem
+        if context['can_vote']:
+            vote = ProblemPointsVote.objects.filter(voter=user.profile, problem=self.object)
+        # whether or not they've already voted
+        context['has_voted'] = context['can_vote'] and vote.exists()
+        if context['has_voted']:
+            context['voted_points'] = vote.first().points  # the previous vote's points
+            context['voted_note'] = vote.first().note
+
+        if 'problem_points_vote_form' not in context:
+            context['problem_points_vote_form'] = ProblemPointsVoteForm({})
+
+        if 'has_errors' not in context:
+            context['has_errors'] = False
+
+        if 'points_placeholder' not in context:  # placeholder for the points
+            if context['has_voted']:  # if voted, the vote
+                context['points_placeholder'] = context['voted_points']
+            else:  # otherwise a *nice* default
+                context['points_placeholder'] = 69.42069
+
+        if 'note_placeholder' not in context:
+            if context['has_voted']:
+                context['note_placeholder'] = context['voted_note']
+            else:
+                context['note_placeholder'] = self.default_note()
+
+        all_votes = sorted([v.points for v in ProblemPointsVote.objects.filter(problem=self.object)])
```

Comment:
```suggestion
        all_votes = list(self.object.problem_points_votes.order_by('points').values_list('points', flat=True))
```
Note: this query is untested, but it should be equivalent and faster since we are not grabbing the entire points object only to use only the `points` field.